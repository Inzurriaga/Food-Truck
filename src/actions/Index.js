export const getCurrentPosition = (position) => ({
    type: "CURRENT_POSITION",
    position
})

export const loadingDisplay = () => ({
    type: "LOADING"
})

export const setTruckList = (trucks) => ({
    type: "SET_TRUCK",
    trucks
})

export const errorHandling = (error) => ({
    type: "ERROR",
    error
})

export const setUser = (userinfo) => ({
    tyep: "SIGN_IN",
    userinfo
})

export const signOut = () => ({
    type: "SIGN_OUT"
})