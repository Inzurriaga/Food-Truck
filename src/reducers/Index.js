import { position } from "./Position"
import { loading } from "./Loading"
import { trucks } from "./Trucks"
import { error } from "./Error"
import { user } from "./User"
import { combineReducers } from 'redux';

export const rootReducer = combineReducers({
    position,
    loading,
    trucks,
    error,
    user
})