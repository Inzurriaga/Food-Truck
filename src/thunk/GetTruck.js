import { loadingDisplay, setTruckList } from "../actions/Index"

export const getTruck = (userInfo) => {
    console.log("hello world", userInfo)
    const option = {
        method: "POST",
        headers: {
            "Content-Type": "application/json"
        },
        body: JSON.stringify(userInfo)
    }
    return async (dispatch) => {
        try {
            dispatch(loadingDisplay())
            const response = await fetch("http://localhost:3000/api/truck", option)
            if (!response.ok) {
                throw new Error(response.statusText)
            }
            const notes = await response.json();
            dispatch(setTruckList(notes));
            dispatch(loadingDisplay());
        } catch (error) {
            console.log(error)
        }
    }
}