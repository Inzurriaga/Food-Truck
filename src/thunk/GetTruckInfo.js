import { loadingDisplay } from "../actions/Index"

export const getTruckInfo = (url) => {
    return async (dispatch) => {
        try {
            dispatch(loadingDisplay())
            const response = await fetch(url)
            if (!response.ok) {
                throw new Error(response.statusText)
            }
            const truck = await response.json();
            dispatch(loadingDisplay());
            return truck
        } catch (error) {
            console.log(error)
        }
    }
}