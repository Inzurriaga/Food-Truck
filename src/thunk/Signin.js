import {} from "../actions/Index"

export const signin = (userType, userInfo) => {
    const option = {
        method: "POST",
        headers: {
            "Content-Type": "application/json"
        },
        body: JSON.stringify({})
    }
    return  async (dispatch) => {
        try {
            const response = await fetch(`http://localhost:3000/api/${userType}/login`, option)
        } catch(error){
            
        }
    }
}