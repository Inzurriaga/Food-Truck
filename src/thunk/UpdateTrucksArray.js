import { loadingDisplay, setTruckList, getCurrentPosition } from "../actions/Index"

export const updateTrucksArray = (distance, address) => {
    console.log(distance, address)
    const option = {
        method: "POST",
        headers: {
            "Content-Type": "application/json"
        },
        body: JSON.stringify({distance, address})
    }
    return async (dispatch) => {
        try {
            dispatch(loadingDisplay())
            const response = await fetch('http://localhost:3000/api/truck/update', option)
            if (!response.ok) {
                throw new Error(response.statusText)
            }
            const trucks = await response.json();
            console.log(trucks)
            dispatch(setTruckList(trucks.trucklist));
            dispatch(getCurrentPosition(trucks.position))
            dispatch(loadingDisplay());
        } catch (error) {
            console.log(error)
        }
    }
}