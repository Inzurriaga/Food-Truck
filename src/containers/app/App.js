import React, { Component } from 'react';
import { connect } from "react-redux";
import { getCurrentPosition, errorHandling } from "../../actions/Index";
import Header from "../../components/header/Header";
import Home from "../home/Home";
import { getTrucksArray } from "../../thunk/GetTrucksArray";
import {  Switch, Route } from "react-router-dom";
import TruckInfo from '../truckInfo/TruckInfo';
import { SignIn } from "../signIn/SignIn"


export class App extends Component {
  componentDidMount(){
    navigator.geolocation.getCurrentPosition(this.getCurrentUserPosition, this.geolocationError)
  }

  getCurrentUserPosition = (position) => {
    const {getTrucksArray, getCurrentPosition} = this.props
    const { latitude, longitude } = position.coords
    const currentPosition = [latitude, longitude]
    const userInfo = {
      distance: 10, 
      latit: latitude,
      long: longitude
    }
    getTrucksArray(userInfo)
    getCurrentPosition(currentPosition)
  }

  geolocationError = () => {
    this.props.errorHandling("GPS not working use the settings to enter a location")
  }
  
  render() {
    return (
      <div className="App">
      <Header />
        <Switch>
          <Route exact path="/" component={Home}/>
          <Route exact path="/signin" component={SignIn}/>
          <Route exact path="/truck/:id" render={({match}) => {
            const { id } = match.params
            return <TruckInfo id={id}/>
          }} />
        </Switch>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  getCurrentPosition: (position) => dispatch(getCurrentPosition(position)),
  getTrucksArray: (userInfo) => dispatch(getTrucksArray(userInfo)),
  errorHandling: (error) => dispatch(errorHandling(error))

})

export const mapStateToProps = (state) => ({
  loading: state.loading,
  trucks: state.trucks
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
