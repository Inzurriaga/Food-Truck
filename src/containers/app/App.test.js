import React from "react";
import { App, mapDispatchToProps, mapStateToProps }  from "./App";
import { getCurrentPosition, errorHandling } from "../../actions/Index";
import { getTrucksArray } from "../../thunk/GetTrucksArray";
import { shallow } from "enzyme";

const mockgetCurrentPosition = jest.fn()
const mockgetTrucksArray = jest.fn()
const mockerrorHandling = jest.fn()

describe("App", () => {
    describe("App", () => {
        let wrapper;
        let mockGeolocation
        beforeEach(() => {
            mockGeolocation = {
                getCurrentPosition: jest.fn(),
                watchPosition: jest.fn()
            };   
            global.navigator.geolocation = mockGeolocation;
            wrapper = shallow(
                <App  getCurrentPosition={mockgetCurrentPosition}
                        getTrucksArray={mockgetTrucksArray}
                        errorHandling={mockerrorHandling}/>
            )
        })
        it("should match snap shot", () => {
            expect(wrapper).toMatchSnapshot()
        })
        it("should call props when getCurrentUserPosition is called", () => {
            wrapper.instance().getCurrentUserPosition()
            expect(mockgetCurrentPosition).toHaveBeenCalled()
            expect(mockgetTrucksArray).toHaveBeenCalled()
        })
        it("should call props when geolocationError is called", () => {
            wrapper.instance().geolocationError()
            expect(mockerrorHandling).toHaveBeenCalled()
        })
    })
    describe("mapDispatchToProps", () => {
        it("should call dispatch with saveHouse", () => {
            const mockDispatch = jest.fn()
            const actionToDispatch = getCurrentPosition([123,123])
            const mappedProps = mapDispatchToProps(mockDispatch)
            mappedProps.getCurrentPosition([123,123])
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
        })
        it("should call dispatch with displayLoading", () => {
            const mockDispatch = jest.fn()
            const actionToDispatch = errorHandling("hello")
            const mappedProps = mapDispatchToProps(mockDispatch)
            mappedProps.errorHandling("hello")
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
        })
        it("should call dispatch with displayLoading", () => {
            const mockDispatch = jest.fn()
            const actionToDispatch = getTrucksArray([123,123])
            const mappedProps = mapDispatchToProps(mockDispatch)
            mappedProps.getTrucksArray([123,123])
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
        })
    })
    describe("mapStateToProps", () => {
        it("should only return a object of what we need from state", () => {
            const mockState = {
                loading: true,
                trucks: [{},{},{}],
                error: "error"
            }
            const expected = {
                loading: true,
                trucks: [{},{},{}]
            }
            const response = mapStateToProps(mockState)
            expect(response).toEqual(expected)
        })
    })
})