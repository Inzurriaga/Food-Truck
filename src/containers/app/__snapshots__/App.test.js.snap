// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should match snap shot 1`] = `
<div
  className="App"
>
  <Header />
  <Switch>
    <ConnectFunction
      exact={true}
      path="/"
    />
    <Route
      exact={true}
      path="/truck/:id"
      render={[Function]}
    />
  </Switch>
</div>
`;
